index.js

```diff
@@ -6,7 +6,11 @@ module.exports = function (CustomEmitter) {
   var emit = Emitter.prototype.emit
 
   function onevent (packet) {
-    var args = packet.data || []
+    var args = [];
+
+    if (packet.data && packet.data.constructor === Array) {
```

Comment:
A socket IO payload should be like [ 'my_event', my_data ]

There is no reason to catch non-array payloads. These are not valid at all.

You can just send raw data over a websocket and crash an entire server.